app/controllers/application_modal_controller.rb

```diff
@@ -0,0 +1,29 @@
+class ApplicationModalController < ApplicationController
```

Comment:
I can move all the code inside ApplicationController. The only reason I created new controller is that I used to extend behaviour by creating subclasses.